### ISSUE TITLE
Author ID no longer working

### ISSUE THREAD

**Issue by WWE3** [cabrerahector/wordpress-popular-posts#157]
As of 3 or 4 updates ago Author ID no longer works as a criteria  in my Popular Post setup. If I put in an author ID, with or without a category number selected, it ignores the author ID and returns the category posts as if there is no Author ID entered. Previous to this Author ID has worked find as a criteria, either in combination with a category or with no category selected. We have used this without problem for several years. The problem began with a Popular Posts update in the last 3-4 months. It was the update where "Custom" was added as date option. When I select "Custom" I am not given a place to enter the "Custom" date so this function does not work either. Thank you for your assistance.

**Comment by cabrerahector** [comment]
Hey there!

Thanks for reporting. I'll look into it as soon as possible.

**Comment by WWE3** [comment]
Thank you for replying! I just updated my comment with more detail. I remembered it was the update where the "Custom" date function was added. Since then Author ID has not worked. Also, when I enter "Custom" as date option no way of selecting the "Custom" date appears.

**Comment by WWE3** [comment]
A few minutes ago I tried selecting "Custom" as the time range option and no entry box appeared. Then I tried manually refreshing the screen. "Custom" remained in the drop down selection box and a data entry box then appeared. So at this point the time range entry box appears only after a manual screen refresh. Still "Author ID" does not work. If I put in an Author ID, the right side of the screen dims as calculations are done, but the entered Author ID makes no difference in posts selected. Sometimes when I try this the author ID # I entered in the Author ID box is no longer in the box after the dimming of the right side of the screen during the calculation. I have attached screen shots, one is after selected Custom time range option, the 2nd is after a manual refresh following selecting the Custom time range option. After either one the Author ID entry does not produce essays by the ID entered, it acts as if no Author ID was entered (this since the update where Custom time range option was added.
![screen shot 2017-10-22 at 10 09 11 am](https://user-images.githubusercontent.com/33002578/31863269-1d9c25ec-b711-11e7-8499-6383249dc6b7.png)
![screen shot 2017-10-22 at 9 22 40 am](https://user-images.githubusercontent.com/33002578/31863270-1daeb4e6-b711-11e7-9955-2e40790d5881.png)

**Comment by cabrerahector** [comment]
Thanks, that makes it much clearer now.

Found out what the problem was and fixed it. An update will follow soon.

> A few minutes ago I tried selecting "Custom" as the time range option and no entry box appeared. Then I tried manually refreshing the screen. "Custom" remained in the drop down selection box and a data entry box then appeared. So at this point the time range entry box appears only after a manual screen refresh.

Regarding this, it's a limitation from WordPress actually. Due to the way widgets are structured it's not possible to dynamically hide / display form elements without resorting to "hacks." I've tried it on another plugin on mine and while it works on the Widgets screen, it doesn't work as nicely on the Customizer.

Things might have changed with the latest versions of WordPress and maybe there's some sort of API that allow developers to create dynamic widgets that I'm not aware of, though. I'll look into this once I get some free time to research and experiment.
